ДЗ11 готово

EvgeniiKovalev opened the pull request:
ДЗ11

EvgeniiKovalev committed:
ДЗ11 готово

EvgeniiKovalev committed:
ДЗ11 добавил вопрос в AppHw11 в строке 13

EvgeniiKovalev committed:
ДЗ11 добавил округление к ближайшему целому в Horse.swim

EvgeniiKovalev committed:
ДЗ11 добавил описание ДЗ

sproshchaev commented on src/main/java/ru/otus/java/basic/homeworks/homework11/AppHw11.java:
```diff
@@ -0,0 +1,73 @@
+
+
+
+        //Tester test = new Tester(); //класс для удобства вывода результатов при запуске методов run, swim при тестирования, не меняя самих методов run, swim
```
здесь и далее: комментарии необходимо удалять перед выполнением pull request. Если какую-то информацию относительно работы метода хочется оставить, то ее можно оформить в виде javadoc (см. пример https://java-online.ru/java-javadoc.xhtml).
В IntelliJ IDEA позволяет оформить javadoc введя /** и нажав Enter

EvgeniiKovalev replied:
убрал комментарии

sproshchaev commented on src/main/java/ru/otus/java/basic/homeworks/homework11/Tester.java:
```diff
@@ -0,0 +1,20 @@
+public class Tester {
+    public static void testRun(TestAction testAction, int distance, Object obj) {
+        if (testAction == TestAction.RUN) {
+            int sec =  ((Animal)obj).run(distance);
```
рекомендуется использовать проверку типа объекта через instanceof для избежания ошибок приведения типов
```
if (obj instanceof Animal) {
    int sec = ((Animal)obj).run(distance);
    // ...
}

```

EvgeniiKovalev replied:
Исправил, а в случае else принято выкидывать исключение какого типа ?

sproshchaev commented on src/main/java/ru/otus/java/basic/homeworks/homework11/Cat.java:
```diff
@@ -0,0 +1,18 @@
+                ", runSpeed=" + runSpeed +
+                ", stamina=" + stamina +
+                ", staminaForRun=" + staminaForRun +
+                '}';
```
лучше использовать String.format() вместо конкатенации строк
```
return String.format("Cat{name='%s', runSpeed=%d, stamina=%d, staminaForRun=%d}", name, runSpeed, stamina, staminaForRun);

```

EvgeniiKovalev replied:
исправил

sproshchaev commented on src/main/java/ru/otus/java/basic/homeworks/homework11/Animal.java:
```diff
@@ -0,0 +1,28 @@
+    protected String name;
+    protected int runSpeed;
+    protected int stamina;
+    protected final int staminaForRun = 1;
```
рекомендуется использовать максимально строгие модификаторы доступа (private) для инкапсуляции данных, чтобы доступ был возможен только через геттеры и сеттеры

EvgeniiKovalev replied:
исправил, применил геттеры и сеттер

EvgeniiKovalev committed:
ДЗ12 исправлены замечания:
  все поля классов стали private
  добавил геттеров и сеттеров
  применил String.format()
  добавил проверку типа через instanceof
